### DISCUSSION TITLE
How to capture meta-mappings (mappings of mappings into SSSOM)

### DISCUSSION THREAD

**Discussion by matentzn** [mapping-commons/sssom#186] [General]
We frequently have the situation where we take an external mapping provided as a spreadsheet (e.g. from NCIT, SNOMED, etc) and wish to convert it to SSSOM for further processing. The question is what to best do about the metamodel mapping. Let's say we have this raw data:

| HPO Term | MPO Term | Mapping |
| ----- | ----- | ----- |
| HP:123 | MP:123 | EXACT |
| HP:124 | MP:124 | BROAD |

It seems wasteful to import this as:

| subject_id | object_id | predicate_id |
| ----- | ----- | ----- |
| HP:123 | MP:123 | someCrazyNamespace:EXACT |
| HP:124 | MP:124 | someCrazyNamespace:BROAD |

Rather we would like something like:

| subject_id | object_id | predicate_id |
| ----- | ----- | ----- |
| HP:123 | MP:123 | skos:exactMatch |
| HP:124 | MP:124 | skos:broadMatch |

Where do we capture that 

someCrazyNamespace:EXACT ---> skos:exactMatch

- 👍  In a separate SSSOM file? Which we somehow link to this one.
- 👀  In the `description field`?
- 🎉  In a new element somewhere?

I notice these kinds of questions are relevant everywhere, like LinkML etc, but just thinking out loud now. Any ideas anyone?

**Comment by jonquet** [comment]
I would not try to capture this.
Here is my 2 cents: 
Producing another SSSOM file with SKOS properties rather than the original ones would means "creating a new set of mappings" with provenance/source the ones produced originally.
Then a bunch of metadata must change.
The commitment/responsibility of going from someCrazyNamespace:EXACT to skos:exactMatch must come with a bunch of new metadata and in best case scenario curation. 
Except if NCI or SNOMED natively adopt these SKOS properties then "standardizing" their data/mappings will mean producing new data.

**Comment by graybeal** [comment]
I would (as a user) try to capture it with the minimal work possible. To me that means translate the original into true SKOS properties (as I think Clement is saying), but figure out a way to say the source for the mappings, and that my exactMatch mappings came from EXACT, and so on (i.e., the translation rules). At least where I could. Or better, I could just point to the software that creates the mappings, which should be in some git repo and which will provide the exact translation methods. 

In either case, you don't have to solve it as part of SSSOM, not part of the standard. That's what best practices are for.

**Comment by matentzn** [comment]
So @graybeal you would merely write a comment or provide a link to the python script to document the translation rules?

**Comment by graybeal** [comment]
I think repeating the rules in the mapping file would be time consuming and error prone. Pointing to the IRI of the conversion software is basic good provenance. So that’s a yes.
________________________________

**Comment by nichtich** [comment]
Real-world data always requires some cleanup and transformation such as renaming column headers and replacing identifiers. It can be done with command line script (e.g. sed or awk) or with search & replace in a spreadsheet software.
